taf/tests/test_updater/test_update/test_update_invalid.py

```diff
@@ -73,3 +79,186 @@ def test_update_invalid_repo_target_in_indeterminate_state(
         UNCOIMITTED_CHANGES,
         True,
     )
+
+
+@pytest.mark.parametrize(
+    "origin_auth_repo",
+    [
+        {
+            "targets_config": [{"name": "target1"}, {"name": "target2"}],
+        },
+    ],
+    indirect=True,
+)
+def test_dirty_index_auth_repo_update_file(origin_auth_repo, client_dir):
+    clone_repositories(
+        origin_auth_repo,
+        client_dir,
+    )
+
+    clients_auth_repo_path = client_dir / origin_auth_repo.name
+    update_file_without_commit(str(clients_auth_repo_path), "dirty_file.txt")
+
+    try:
+        # Attempt to update, which should fail due to uncommitted changes
+        update_and_check_commit_shas(
+            OperationType.UPDATE,
+            origin_auth_repo,
+            client_dir,
+        )
+        assert False, "Update should have failed due to uncommitted changes"
+    except UpdateFailedError as e:
+        assert f"Unexpected error: {e}"
+    update_and_check_commit_shas(
+        OperationType.UPDATE,
+        origin_auth_repo,
+        client_dir,
+        force=True,
+    )
+
+
+@pytest.mark.parametrize(
+    "origin_auth_repo",
+    [
+        {
+            "targets_config": [{"name": "target1"}, {"name": "target2"}],
+        },
+    ],
+    indirect=True,
+)
+def test_dirty_index_auth_repo_add_file(origin_auth_repo, client_dir):
+    clone_repositories(
+        origin_auth_repo,
+        client_dir,
+    )
+
+    auth_repo_path = origin_auth_repo.path / "namespace/auth_repo"
+    add_file_without_commit(str(auth_repo_path), "new_file.txt")
```

Comment:
Should be updated like the previous one - update client's repo and specify which error should be raised